core/controllers/acl_decorators.py

```diff
@@ -2615,3 +2617,140 @@ def test_can_play_entity(self, entity_type, entity_id, **kwargs):
+            raise base.UserFacingExceptions.NotLoggedInException
+
+        return handler(self, **kwargs)
+
+    test_can_view_submitted_voiceover_applications.__wrapped__ = True
+
+    return test_can_view_submitted_voiceover_applications
+
+
+def can_review_voiceover_application(handler):
+    """Decorator to check whether the user can review voiceover applications.
+
+    Args:
+        handler: function. The function to be decorated.
+
+    Returns:
+        function. The newly decorated function that now also checks if a
+            user has permission to review voiceover application.
+    """
+
+    def test_can_review_voiceover_application(
+            self, **kwargs):
+        """Checks if the user can review voiceover application.
+
+        Returns:
+            *. The return value of the decorated function.
+
+        Raises:
+            NotLoggedInException: The user is not logged in.
+            UnauthorizedUserException: The user does not have
+                credentials to voiceover an exploration.
+        """
+        if not self.user_id:
+            raise base.UserFacingExceptions.NotLoggedInException
+
+        user_actions_info = user_services.UserActionsInfo(self.user_id)
+        if (
+                role_services.ACTION_ACCEPT_ANY_VOICEOVER_APPLICATION not in
+                user_actions_info.actions):
+            raise self.UnauthorizedUserException(
+                'You do not have credentials to review voiceover '
+                'applications.')
+
+        return handler(self, **kwargs)
+
+    test_can_review_voiceover_application.__wrapped__ = True
+
+    return test_can_review_voiceover_application
+
+
+def can_submit_voiceover_application(handler):
+    """Decorator to check whether a user can submit voiceover application.
+
+    Args:
+        handler: function. The function to be decorated.
+
+    Returns:
+        function. The newly decorated function that now also checks
+            if the user has permission to submit voiceover application.
+    """
+
+    def test_can_submit_voiceover_application(self, **kwargs):
+        """Checks if the user can submit voiceover application.
+
+        Args:
+            **kwargs: *. Keyword arguments.
+
+        Returns:
+            *. The return value of the decorated function.
+
+        Raises:
+            UnauthorizedUserException: The user does not have
+                credentials to submit voiceover application.
+        """
+        if role_services.ACTION_SUBMIT_VOICEOVER_APPLICATION in (
+                self.user.actions):
+            # The voiceover_service returns user submitted voiceover
+            # applications in sorted order of the date of submission.
+            voiceover_applications = (
+                voiceover_services.get_user_submitted_voiceover_applications(
+                    self.user_id))
+            if voiceover_applications:
+                latest_voiceover_application = voiceover_applications[0]
```

Comment:
There can be more than one?

Reply:
Yeah, a user can have multiple applications on their name, we are checking the status of the last application.

Reply:
I thought the user can only have one active application at a time?

Otherwise, if one application is accepted, what happens to the rest?